### DISCUSSION TITLE
March 2024 meeting

### DISCUSSION THREAD

**Discussion by Eothred** [PyORBIT-Collaboration/PyORBIT3#11] [General]
We met on March 26th, 2024, to discuss the evaluation of scikit-build and meson tests, as well as other ongoing developments

### Meeting notes

- We discussed pull requests and closed  https://github.com/PyORBIT-Collaboration/PyORBIT3/pull/7
- We discussed different options for incremental build systems
  - cmake based (scikit-build-core) and meson based
  - agreed that meson looks better if we can figure out
    - pure python packages
    - system independent discovery of libraries (fftw)
- Discussed potential discrepancies for the ring simulations.

### TODO

- Yngve will schedule the next meeting  in ~ 1 month
- Juan will make a test that checks the new B_rho for ions and corrector polarities https://github.com/PyORBIT-Collaboration/PyORBIT3/pull/8
- Sasha will send out a link to the latest developments with meson build system.

### Relevant links

- meson build development can be found on the branch https://github.com/azukov/PyORBIT3/tree/meson
- scikit-build-core demo can be found on the branch https://github.com/Eothred/PyORBIT3/tree/test_cmake_build_style

**Comment by azukov** [comment]
Just updated my [meson](https://github.com/azukov/PyORBIT3/tree/meson) branch with something that actually builds full package including pure python. It runs full SNS linac example now.

**Reply by Eothred** [reply]
Thanks. I had a look and could build after some minor manual edits at least. Maybe you could open a [draft merge request](https://github.blog/2019-02-14-introducing-draft-pull-requests/) and we could start a discussion there?

Edit: for anyone curious who are on a Linux system, [here are the fixes which worked for me](https://github.com/Eothred/PyORBIT3/commit/fd7abcbe45aa296aef3a177141d59ec101777f70).
